tests/integration/telemetry/policy/envoy_ratelimit_test.go

```diff
@@ -0,0 +1,174 @@
+// +build integ
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/echo/common/response"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/istio/ingress"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/util/tmpl"
+)
+
+var (
+	ist         istio.Instance
+	echoNsInst  namespace.Instance
+	ratelimitNs namespace.Instance
+	ing         ingress.Instance
+	srv         echo.Instance
+	clt         echo.Instance
+)
+
+func TestRateLimiting_DefaultLessThanOverride(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			if !sendTrafficAndCheckIfRatelimited(t) {
+				t.Errorf("No request received StatusTooMantRequest Error.")
+			}
+		})
+}
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		RequireSingleCluster().
+		Label(label.CustomSetup).
+		Setup(istio.Setup(&ist, nil)).
+		Setup(testSetup).
+		Run()
+}
+
+func testSetup(ctx resource.Context) (err error) {
+	echoNsInst, err = namespace.New(ctx, namespace.Config{
+		Prefix: "istio-echo",
+		Inject: true,
+	})
+	if err != nil {
+		return
+	}
+
+	_, err = echoboot.NewBuilder(ctx).
+		With(&clt, echo.Config{
+			Service:   "clt",
+			Namespace: echoNsInst}).
+		With(&srv, echo.Config{
+			Service:   "srv",
+			Namespace: echoNsInst,
+			Ports: []echo.Port{
+				{
+					Name:     "http",
+					Protocol: protocol.HTTP,
+					// We use a port > 1024 to not require root
+					InstancePort: 8888,
+				},
+			}}).
+		Build()
+	if err != nil {
+		return
+	}
+
+	ing = ist.IngressFor(ctx.Clusters().Default())
+
+	ratelimitNs, err = namespace.New(ctx, namespace.Config{
+		Prefix: "istio-ratelimit",
+	})
+	if err != nil {
+		return
+	}
+
+	yamlContent, err := ioutil.ReadFile("testdata/ratelimitservice.yaml")
+	if err != nil {
+		return
+	}
+
+	err = ctx.Config().ApplyYAML(ratelimitNs.Name(),
```

Comment:
seems like nothing is removing this when the test is complete?

Reply:
It will get deleted as part of ratelimit ns cleanup at the end of the test...